tests/integration/test_logcollector/test_macos/test_macos_file_status_basic.py

```diff
@@ -0,0 +1,118 @@
+from wazuh_testing.remote import check_agent_received_message
+from wazuh_testing.tools.monitoring import wait_file
+from wazuh_testing.tools.file import read_json
+from time import sleep
+
+# Marks
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=0)]
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_macos_file_status_basic.yaml')
+
+parameters = [{'ONLY_FUTURE_EVENTS': 'yes'}, {'ONLY_FUTURE_EVENTS': 'no'}]
+metadata = [{'only-future-events': 'yes'}, {'only-future-events': 'no'}]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f'{x["ONLY_FUTURE_EVENTS"]}' for x in parameters]
+
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = {'logcollector.vcheck_files': str(file_status_update_time)}
+
+
+@pytest.fixture(scope='module')
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+# Fixtures
+@pytest.fixture(scope='module', params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def get_connection_configuration():
+    """Get configurations from the module."""
+    return logcollector.DEFAULT_AUTHD_REMOTED_SIMULATOR_CONFIGURATION
+
+
+def extra_configuration_before_yield():
+    """Delete file status file."""
+    os.remove(LOGCOLLECTOR_FILE_STATUS_PATH) if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH) else None
+
+
+def test_macos_file_status_basic(get_local_internal_options, configure_local_internal_options, get_configuration,
+                                 configure_environment, get_connection_configuration, init_authd_remote_simulator,
+                                 restart_logcollector):
+
+    """Checks if logcollector stores correctly "macos"-formatted localfile data.
+
+    This test uses logger tool and a custom log to generate an ULS event. The agent is connected to the authd simulator
+    and sends an event to trigger the file_status.json update.
+
+    Raises:
+        TimeoutError: If the callbacks, that checks the expected logs, are not satisfied in the expected time.
+        FileNotFoundError: If the file_status.json is not available in the expected time.
+    """
+
+    macos_message = {
+        'command': 'logger',
+        'message': 'Logger testing message - file status',
+    }
+
+    macos_logcollector_monitored = logcollector.callback_monitoring_macos_logs
+    wazuh_log_monitor.start(timeout=15, callback=macos_logcollector_monitored,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_TARGET_SOCKET)
+
+    # Waits 20 seconds to give time to logcollector to start the module
+    sleep(20)
```

Comment:
Hardcoded sleeps aren't allowed either, we need to create a variable for the sleep that explains why it is needed